cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
@vadz As I already mentioned in #390 option -Wno-pedantic-ms-format make no sense for compilers with version greater or equal 4.7.1 
So, to fix this fast - I remove only for Windows this check. For Travis-CI it's still enforced.

Another option I can suggest - provide macro in soci-compiler.h (https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html)
``` 
#define GCC_VERSION (__GNUC__ * 10000 \
                               + __GNUC_MINOR__ * 100 \
                               + __GNUC_PATCHLEVEL__)

```
and do check something like this

```
#if defined(_MSC_VER) || ( defined(__MINGW32__) && (GCC_VERSION < 40701) )
```
This should work.